### PR TITLE
[system-probe] Add and reorganize stats

### PR DESCRIPTION
**Pull request by hmahmood**
### What does this PR do?

Adds the following stats for system-probe:
- gateway lookup/route cache
- breakdown of http hits by status code
- connection count broken down by protocol and ipv4/6

Also reorganize how stats are collected for uniformity.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/http/telemetry.go [hunk 1]
```diff
@@ -44,41 +64,32 @@ func (t *telemetry) aggregate(txs []httpTX, err error) {
...
+		float64(totalRequests)/float64(elapsed),
+		misses,
+		float64(misses)/float64(elapsed),
+		dropped,
+		float64(rejected)/float64(elapsed),
+		rejected,
+		float64(dropped)/float64(elapsed),
+		aggregations,
 	)
+
```
Comment (derekwbrown) [inline pkg/network/http/telemetry.go:93]: For this function, it seems like we should check the log level first.
The `misses`, `dropped`, etc. variables are assigned, and then the 4 floating point conversions and divisions are done regardless of log level.  if the entire function below `if stats:= .....` was `if loglevel >= debug`, then all of those computations wouldn't have to be done on every execution.

**Review by derekwbrown — APPROVED** [review]

File: releasenotes/notes/npm-udpv6-receive-1a62cf8b6080b7c6.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Add missing support for UDPv6 receive path to NPM
```
Comment (maycmlee) [inline releasenotes/notes/npm-udpv6-receive-1a62cf8b6080b7c6.yaml:4]:
```suggestion
    Add missing support for UDPv6 receive path to NPM.
```

**Review by maycmlee — APPROVED** [review]
Small edit

File: releasenotes/notes/npm-udpv6-receive-1a62cf8b6080b7c6.yaml (hunk 2 above)
Comment (maycmlee) [inline releasenotes/notes/npm-udpv6-receive-1a62cf8b6080b7c6.yaml:4]:
```suggestion
    Add missing support for UDPv6 receive path to NPM.
```

**Review by maycmlee — APPROVED** [review]
Small edit

**Pull request merged**